src/codeu/chat/server/Server.java

```diff
@@ -57,19 +61,32 @@
   private final Uuid id;
   private final Secret secret;
 
-  private final Model model = new Model();
-  private final View view = new View(model);
   private final Controller controller;
 
   private final Relay relay;
   private Uuid lastSeen = Uuid.NULL;
 
-  public Server(final Uuid id, final Secret secret, final Relay relay) {
+  private final Model model;
+  final View view;
+
+  public Server(final Uuid id, final Secret secret, final Relay relay, final File dir) {
 
     this.id = id;
     this.secret = secret;
-    this.controller = new Controller(id, model);
     this.relay = relay;
+    File fp = new File(dir, "chatLog.dat");
+    if (fp == null) {
```

Comment:
Sorry, if it *doesn't* exist you should run the if statement.

`if( fp == null && !File.exists(fp)) {`